bin/sozo/src/ops/register.rs

```diff
@@ -1,25 +1,59 @@
+use std::collections::HashMap;
+
 use anyhow::{Context, Result};
 use dojo_world::contracts::WorldContract;
 use dojo_world::metadata::Environment;
 use dojo_world::utils::TransactionWaiter;
+use scarb::core::Config;
 use starknet::accounts::Account;
+use starknet_crypto::FieldElement;
 
 use crate::commands::register::RegisterCommand;
 
-pub async fn execute(command: RegisterCommand, env_metadata: Option<Environment>) -> Result<()> {
+pub async fn execute(
+    command: RegisterCommand,
+    env_metadata: Option<Environment>,
+    config: &Config,
+) -> Result<()> {
     match command {
         RegisterCommand::Model { models, world, starknet, account, transaction } => {
             let world_address = world.address(env_metadata.as_ref())?;
             let provider = starknet.provider(env_metadata.as_ref())?;
 
             let account = account.account(&provider, env_metadata.as_ref()).await?;
             let world = WorldContract::new(world_address, &account);
+            let mut model_class_hashes = HashMap::new();
+            for model_class_hash in models.iter() {
+                let (class_hash, contract_address) =
+                    world.model(model_class_hash.into()).call().await?;
+                if class_hash.0 != FieldElement::from_hex_be("0x0")?
+                    && contract_address.0 != FieldElement::from_hex_be("0x0")?
+                {
+                    model_class_hashes.insert(model_class_hash, true);
+                }
+            }
+
+            }
 
             let calls = models
                 .iter()
+                .filter(|m| {
+                    if model_class_hashes.contains_key(m) {
+                        config.ui().print(format!(
+                            "\"{:?}\" model already registered with the given class hash", m
+                        ));
+                        return false;
+                    }
+                    true
+                })
```

Comment:
I think this filtering isn't necessary considering we can use the `for` loop above to filter out only the models we need to register.